pandas/core/indexes/base.py

```diff
@@ -1949,6 +1949,9 @@ def __nonzero__(self):
     def __contains__(self, key):
         hash(key)
         try:
+            if is_float(key) and is_integer_dtype(self.dtype):
+                if key != int(key):
+                    return False
```

Comment:
I would put this outside of the try/except block

Reply:
done

Reply:
@jorisvandenbossche I think it can't pass the test!

Reply:
So it fails when key is NaN ?

Reply:
I think so. The block is needed.